File: backend/Service/src/main/java/ru/nikidzawa/db/store/dto/object/dto/MessageDto.java
package ru.nikidzawa.db.store.dto.object.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MessageDto {
    Long id;
    LocalDateTime createdAt;
    String text;
    Long senderId;
}

File: frontend/react-app/src/api/dto/MessageDto.js
export default class MessageDto {
    constructor(id, createdAt, text, senderId) {
        this.id = id;
        this.createdAt = new Date(createdAt);
        this.text = text;
        this.senderId = senderId;
    }

    static fromJSON(data) {
        return new MessageDto(
            data.id,
            data.createdAt,
            data.text,
            data.senderId
        );
    }
}
